anta/catalog.py

```diff
@@ -328,40 +334,59 @@ def from_list(data: ListAntaTestTuples) -> AntaCatalog:
             raise
         return AntaCatalog(tests)
 
-    def get_tests_by_tags(self, tags: set[str], *, strict: bool = False) -> list[AntaTestDefinition]:
-        """Return all the tests that have matching tags in their input filters.
+    def build_indexes(self, filtered_tests: set[str] | None = None) -> None:
```

Comment:
Could we build the indexes in the AntaCatalog constructor ?
This would allow us to get rid of the boolean.